Add currency conversion to maxFeePerGas > baseFee check

karlb opened the pull request:
We missed this check when adding the CIP-64 tx.

karlb committed:
CIP-64: Convert baseFee for GasFeeCap check

karlb committed:
Add a second FeeCurrency to dev mode genesis

This one has a conversion rate with a smaller value than CELO. Using
both currencies in tests allows to detect missing or wrong currency
conversions.
It might make sense to rename both fee currencies to something like
`DevFeeCurrencyAddr`.

karlb committed:
Test conv-rate in baseFee < MaxFeePerGas check

karlb committed:
Fix e2e test after fixing CIP-64 base fee check

palango reviewed (approved):
Code looks good and nicely tested.

I agree with the `DevFeeCurrencyAddr` renaming, feel free to go ahead with this if you want in this PR.

karlb committed:
Rename FeeCurrencyAddr to DevFeeCurrencyAddr

These are only available in `--dev` mode and the previous naming sounded
too generic.

karlb committed:
Use gh action to install foundry and pin version

It is better to explicitly update the foundry version in a separate PR
rather than having you PR suddenly fail because a new foundry version
has been published.
